pkg/controllers/jobset_controller.go

```diff
@@ -309,9 +425,9 @@ func (r *JobSetReconciler) shouldCreateJob(jobName string, ownedJobs *childJobs)
 	return true
 }
 
-// updateStatus updates the status of a JobSet by appending
+// updateStatusCondition updates the status of a JobSet by appending
 // the new condition to jobset.status.conditions.
-func (r *JobSetReconciler) updateStatus(ctx context.Context, js *jobset.JobSet, condition metav1.Condition) error {
+func (r *JobSetReconciler) updateStatusCondition(ctx context.Context, js *jobset.JobSet, condition metav1.Condition) error {
```

Comment:
always appending is fine for now since we only update one condition that is terminal, but when supporting cases like suspend, we need to first check if the condition already exists and update it in place if necessary

Reply:
Ok, noted.